HW1/P4/P4.txt

```diff
@@ -1 +1,18 @@
-# Your discussion here
+1. Description:
+Firstly, I set up a RDD with the key of each comic issue, the value of the characters that appear in that comic issue. Through several transformations, I set up a final graph, which is a paired RDD, with the key of each character, and the value of all the characters who are in the same comic issue.
+
+2. Graph diameter:
+We start at distance 0 which can be seen as the center of the graph. Every time we move to the neighbours with distance of 1. Since the diameter is 10, meaning that we can move 5 times. Therefore, the maximum number of steps we can search is 5.
+
```

Comment:
Hi chengcheng, since the diameter of the graph is the greatest distance between any two vertices in a graph. https://en.wikipedia.org/wiki/Distance_(graph_theory), I think the worst case scenario would be 10, not 5.